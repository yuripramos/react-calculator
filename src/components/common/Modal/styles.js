import styled from "styled-components";
import { rem } from "../../../styles/tools";
import { red, black50, white, grey20 } from "../../../styles/settings";

export const Title = styled.h1`
  color: ${red};
  font-family: "OpenSans Thin";
  font-size: ${rem(16)};
  letter-spacing: ${rem(0.3)};
  margin-bottom: ${rem(18)};
`;

export const IconContainer = styled.div`
  color: ${red};
  margin-bottom: ${rem(10)};
`;

export const Description = styled.p`
  font-family: Lato;
  font-size: ${rem(13)};
  line-height: 1.46;
  letter-spacing: ${rem(0.3)};
  color: ${black50};
  margin: 0 ${rem(30)};
`;

export const Close = styled.span`
  font-family: "OpenSans Thin";
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.2)};
  text-align: left;
  margin-left: ${rem(10)};
`;

export const CloseBtn = styled.a`
  height: ${rem(43)};
  display: flex;
  align-items: center;
  justify-content: center;
  border-top: solid ${rem(0.7)} ${grey20};
  color: ${black50};
  cursor: pointer;
  margin-top: ${rem(18)};
`;

export const Content = styled.section`
  position: relative;
  z-index: 1;
  border-radius: ${rem(8)};
  background-color: ${white};
  box-shadow: 0 ${rem(4)} ${rem(40)} ${rem(10)} rgba(0, 0, 0, 0.2);
  padding: ${rem(20)};
`;

export const Wrapper = styled.div`
  position: fixed;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 100;
  text-align: center;
  top: 0;
  left: 0;
`;
