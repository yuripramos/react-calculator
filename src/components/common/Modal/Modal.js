import React from "react";
import { string, func, oneOfType, bool, shape, arrayOf } from "prop-types";
import { Overlay } from "../../../styles/objects";
/*eslint-disable*/
/*Modal COMPS*/
import DefaultModal from "./Default";

import { Wrapper, Content } from "./styles";



export const MODAL_TYPES = {
  DEFAULT: "Default"
};

const MODAL_COMPONENTS = {
  Default: DefaultModal
};

function Modal(props) {
  if (props.modalSettings && props.modalSettings.isOpen) {
    const { type } = props.modalSettings;

    const ModalContent =
      MODAL_COMPONENTS[type] || MODAL_COMPONENTS[MODAL_TYPES.DEFAULT];

    return (
      <Wrapper>
        <Content>
          <ModalContent {...props} />
        </Content>
        <Overlay data-test="Overlay" />
      </Wrapper>
    );
  }

  return null;
}

Modal.defaultProps = {
  modal: null
};

Modal.propTypes = {
  modal: oneOfType([
    shape({
      isOpen: bool,
      type: string,
      title: string,
      icon: string,
      description: oneOfType([arrayOf(string), string]),
      onClose: func
    })
  ])
};

export default Modal;
