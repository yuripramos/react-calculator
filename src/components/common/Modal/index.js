import { connect } from "redux-zero/react";

import Modal from "./Modal";
import actions from "../../../actions/modal";

export default connect(({ modalSettings }) => ({ modalSettings }), actions)(
  Modal
);
