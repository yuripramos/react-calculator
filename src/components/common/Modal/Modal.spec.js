import React from "react";
import { render } from "enzyme";

import Modal from "./Modal";

describe("Modal component", () => {
  it("should match snapshot with a closed modal", () => {
    expect(render(<Modal />)).toMatchSnapshot();
  });

  it("should match snapshot with an open modal", () => {
    const props = {
      modalSettings: {
        isOpen: true
      }
    };
    expect(render(<Modal {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with an open modal with title", () => {
    const props = {
      modalSettings: {
        title: "title",
        isOpen: true
      }
    };
    expect(render(<Modal {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with an open modal with icon", () => {
    const props = {
      modalSettings: {
        icon: "close",
        isOpen: true
      }
    };
    expect(render(<Modal {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with an open modal with description string", () => {
    const props = {
      modalSettings: {
        description: "description",
        isOpen: true
      }
    };
    expect(render(<Modal {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with an open modal with description array", () => {
    const props = {
      modalSettings: {
        description: ["description1", "description2"],
        isOpen: true
      }
    };
    expect(render(<Modal {...props} />)).toMatchSnapshot();
  });
});
