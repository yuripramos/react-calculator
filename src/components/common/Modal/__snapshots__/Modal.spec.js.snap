// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should match snapshot with a closed modal 1`] = `null`;

exports[`Modal component should match snapshot with an open modal 1`] = `
<div
  class="sc-htoDjs kwXCJw"
>
  <section
    class="sc-gzVnrw dtKkJq"
  >
    <div
      class="sc-dnqmqq geJMwm"
    >
      <a
        class="sc-bZQynM divcNJ"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.12 15.24A7.12 7.12 0 1 0 8.12 1a7.12 7.12 0 0 0 0 14.24zm0-.95a6.17 6.17 0 1 1 0-12.34
 6.17 6.17 0 0 1 0 12.34zm-.084-6.639L6.19 5.804a.475.475 0 0 0-.672.671l1.848 1.848-1.848
 1.847a.475.475 0 0 0 .672.671l1.847-1.847
 1.848 1.847a.475.475 0 0 0 .67-.671L8.708 8.323l1.848-1.848a.475.475 0 0 0-.671-.671L8.036 7.65z"
            fill="currentColor"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-width=".4"
          />
        </svg>
        <span
          class="sc-EHOje AkWcR"
        >
          Close
        </span>
      </a>
    </div>
  </section>
  <div
    class="sc-bwzfXH dXmOMB"
    data-test="Overlay"
  />
</div>
`;

exports[`Modal component should match snapshot with an open modal with description array 1`] = `
<div
  class="sc-htoDjs kwXCJw"
>
  <section
    class="sc-gzVnrw dtKkJq"
  >
    <div
      class="sc-dnqmqq geJMwm"
    >
      <p
        class="sc-ifAKCX doFnNx"
      >
        description1
      </p>
      <p
        class="sc-ifAKCX doFnNx"
      >
        description2
      </p>
      <a
        class="sc-bZQynM divcNJ"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.12 15.24A7.12 7.12 0 1 0 8.12 1a7.12 7.12 0 0 0 0 14.24zm0-.95a6.17 6.17 0 1 1 0-12.34
 6.17 6.17 0 0 1 0 12.34zm-.084-6.639L6.19 5.804a.475.475 0 0 0-.672.671l1.848 1.848-1.848
 1.847a.475.475 0 0 0 .672.671l1.847-1.847
 1.848 1.847a.475.475 0 0 0 .67-.671L8.708 8.323l1.848-1.848a.475.475 0 0 0-.671-.671L8.036 7.65z"
            fill="currentColor"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-width=".4"
          />
        </svg>
        <span
          class="sc-EHOje AkWcR"
        >
          Close
        </span>
      </a>
    </div>
  </section>
  <div
    class="sc-bwzfXH dXmOMB"
    data-test="Overlay"
  />
</div>
`;

exports[`Modal component should match snapshot with an open modal with description string 1`] = `
<div
  class="sc-htoDjs kwXCJw"
>
  <section
    class="sc-gzVnrw dtKkJq"
  >
    <div
      class="sc-dnqmqq geJMwm"
    >
      <p
        class="sc-ifAKCX doFnNx"
      >
        description
      </p>
      <a
        class="sc-bZQynM divcNJ"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.12 15.24A7.12 7.12 0 1 0 8.12 1a7.12 7.12 0 0 0 0 14.24zm0-.95a6.17 6.17 0 1 1 0-12.34
 6.17 6.17 0 0 1 0 12.34zm-.084-6.639L6.19 5.804a.475.475 0 0 0-.672.671l1.848 1.848-1.848
 1.847a.475.475 0 0 0 .672.671l1.847-1.847
 1.848 1.847a.475.475 0 0 0 .67-.671L8.708 8.323l1.848-1.848a.475.475 0 0 0-.671-.671L8.036 7.65z"
            fill="currentColor"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-width=".4"
          />
        </svg>
        <span
          class="sc-EHOje AkWcR"
        >
          Close
        </span>
      </a>
    </div>
  </section>
  <div
    class="sc-bwzfXH dXmOMB"
    data-test="Overlay"
  />
</div>
`;

exports[`Modal component should match snapshot with an open modal with icon 1`] = `
<div
  class="sc-htoDjs kwXCJw"
>
  <section
    class="sc-gzVnrw dtKkJq"
  >
    <div
      class="sc-dnqmqq geJMwm"
    >
      <div
        class="sc-bxivhb cRcMeL"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.12 15.24A7.12 7.12 0 1 0 8.12 1a7.12 7.12 0 0 0 0 14.24zm0-.95a6.17 6.17 0 1 1 0-12.34
 6.17 6.17 0 0 1 0 12.34zm-.084-6.639L6.19 5.804a.475.475 0 0 0-.672.671l1.848 1.848-1.848
 1.847a.475.475 0 0 0 .672.671l1.847-1.847
 1.848 1.847a.475.475 0 0 0 .67-.671L8.708 8.323l1.848-1.848a.475.475 0 0 0-.671-.671L8.036 7.65z"
            fill="currentColor"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-width=".4"
          />
        </svg>
      </div>
      <a
        class="sc-bZQynM divcNJ"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.12 15.24A7.12 7.12 0 1 0 8.12 1a7.12 7.12 0 0 0 0 14.24zm0-.95a6.17 6.17 0 1 1 0-12.34
 6.17 6.17 0 0 1 0 12.34zm-.084-6.639L6.19 5.804a.475.475 0 0 0-.672.671l1.848 1.848-1.848
 1.847a.475.475 0 0 0 .672.671l1.847-1.847
 1.848 1.847a.475.475 0 0 0 .67-.671L8.708 8.323l1.848-1.848a.475.475 0 0 0-.671-.671L8.036 7.65z"
            fill="currentColor"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-width=".4"
          />
        </svg>
        <span
          class="sc-EHOje AkWcR"
        >
          Close
        </span>
      </a>
    </div>
  </section>
  <div
    class="sc-bwzfXH dXmOMB"
    data-test="Overlay"
  />
</div>
`;

exports[`Modal component should match snapshot with an open modal with title 1`] = `
<div
  class="sc-htoDjs kwXCJw"
>
  <section
    class="sc-gzVnrw dtKkJq"
  >
    <div
      class="sc-dnqmqq geJMwm"
    >
      <h1
        class="sc-htpNat emkUyl"
      >
        title
      </h1>
      <a
        class="sc-bZQynM divcNJ"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.12 15.24A7.12 7.12 0 1 0 8.12 1a7.12 7.12 0 0 0 0 14.24zm0-.95a6.17 6.17 0 1 1 0-12.34
 6.17 6.17 0 0 1 0 12.34zm-.084-6.639L6.19 5.804a.475.475 0 0 0-.672.671l1.848 1.848-1.848
 1.847a.475.475 0 0 0 .672.671l1.847-1.847
 1.848 1.847a.475.475 0 0 0 .67-.671L8.708 8.323l1.848-1.848a.475.475 0 0 0-.671-.671L8.036 7.65z"
            fill="currentColor"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-width=".4"
          />
        </svg>
        <span
          class="sc-EHOje AkWcR"
        >
          Close
        </span>
      </a>
    </div>
  </section>
  <div
    class="sc-bwzfXH dXmOMB"
    data-test="Overlay"
  />
</div>
`;
